### PR TITLE
Added bison parser and flex lexer working with cmake

### LINKED ISSUE #3: Research parser generator
Do initial research on parser generators. Look at https://www.gnu.org/software/bison/

### PR DESCRIPTION
**Pull request by Willu12**
I think we should add lib for CLI

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -10,7 +10,7 @@ add_compile_options(-Wall -Wextra -Wpedantic)
 find_package(GTest CONFIG REQUIRED)
 
 # using compiler cache to speed up build
-set(USE_CCACHE YES)
+#set(USE_CCACHE YES)
```
Comment (jkurdek) [inline CMakeLists.txt]: Why is this commented out?

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -1,3 +1,29 @@
...
+set(BISON_INPUT ${CMAKE_SOURCE_DIR}/libs/parser/parser.y)
+set(FLEX_INPUT ${CMAKE_SOURCE_DIR}/libs/lexer/lexer.l)
+
+# Bison target - generates parser.cpp and parser.hpp
+BISON_TARGET(MyParser ${BISON_INPUT} ${CMAKE_SOURCE_DIR}/libs/parser/parser.cpp COMPILE_FLAGS "--defines=${CMAKE_SOURCE_DIR}/libs/parser/parser.hpp")
+
+# Flex target - generates lexer.cpp
+FLEX_TARGET(MyLexer ${FLEX_INPUT} ${CMAKE_SOURCE_DIR}/libs/lexer/lexer.cpp)
+
+# Ensure Bison is run before Flex
```
Comment (jkurdek) [inline src/CMakeLists.txt:16]: Not the other way around? I think Flex should run first

**Review by jkurdek — APPROVED** [review]
LGTM!

File: src/CMakeLists.txt (hunk 2 above)
Comment (Willu12) [reply src/CMakeLists.txt:16]: Flex scanners often use tokens defined by Bison: the code generated by Flex depends of the header generated by Bison. 

from cmake docs

**Pull request merged**